library/datasort.c

```diff
@@ -560,6 +561,80 @@ static int datasort_split_iterator_free(struct eblob_iterate_control *ictl __att
 	return 0;
 }
 
+static int datasort_load_index(struct datasort_cfg *dcfg,
+			       struct eblob_base_ctl *bctl,
+			       struct datasort_chunk *chunk) {
+	// Sanity
+	assert(dcfg);
+	assert(bctl);
+	assert(chunk);
+	assert(chunk->index == NULL); // Assume that index is already empty and we don't need clear previous data
+
+	int err;
+	struct eblob_file_ctl index_ctl = bctl->index_ctl;
```

Comment:
Why is it copied?

Reply:
From definition of structure we can conclude that this is just a copy of 4 ints.
https://github.com/s-mx/eblob/blob/master/library/blob.h#L93

Reply:
Let's not copy whole structures unless it is required.